### DISCUSSION TITLE
Q: How to adjust the picture orientation

### DISCUSSION THREAD

**Discussion by bdim404** [Sped0n/bridget#363] [Q&A] [answered]
I used your theme to build my own site, but some of the images I uploaded are clearly vertical, but the site will show them rotated to landscape.

Example:
This is the photo what i upload:
![DSC09624](https://github.com/user-attachments/assets/87bca8dc-9864-4d5f-8f99-fd7d143673e9)

This is the what it looks on my site:
<img width="1447" alt="图片" src="https://github.com/user-attachments/assets/d1fa3f75-5864-490d-bb84-5eff6d24531e">

So... How can I solve this problems?

**Comment by 713b76b7-2247-4bd5-98a7-014226363f9b** [comment]
See https://discourse.gohugo.io/t/image-exif-orientation/22902 and https://discourse.gohugo.io/t/why-is-hugo-rotating-my-image/23329.

**Reply by bdim404** [reply]
Thanks for the answer, I read the link you sent.

I am using your theme to build a separate site, and directly adjusted some of your parts of it and then used it directly, I am not good at front-end code, and I know that it is needed to hugo read the orientation information in the exif information. But I don't know how to do that, so after I read it I still don't know how I should fix the image orientation issue.

Can you provide me with a sample modification if possible?

**Reply by 713b76b7-2247-4bd5-98a7-014226363f9b** [reply]
Image parsing in hugo is built in, and I cannot do anything about it. 

Maybe you can check whether the exif info is corrupted, if it is, try to repair it and build the site again.

**Reply by bdim404** [reply]
Wow, I see, it seems to me that you're using the image processed by hugo here in `layouts/partials/function/getImageSlice.html`?

If so, I can try to modify the contents of this file to try to read the `orientation` value of the exif and use it to assist in straightening the image.

My EXIF info is fine, I checked it with `exiftool`.

Thanks for your reply!

**Reply by 713b76b7-2247-4bd5-98a7-014226363f9b** [reply]
> Wow, I see, it seems to me that you're using the image processed by hugo here in layouts/partials/function/getImageSlice.html?

That's processing, what I mean is parsing. Bridget will only scale your image.

> If so, I can try to modify the contents of this file to try to read the orientation value of the exif and use it to assist in straightening the image.

Yes you can, if I remember correctly, there is a code snippet in one of the link I provided above.

**Reply by gustyforcast** [reply]
I also had this problem.
With the way that you solved it did it require handling it within that getImageSlice.html?
Would it be possible to add that update to the project?

**Reply by 713b76b7-2247-4bd5-98a7-014226363f9b** [reply]
> Would it be possible to add that update to the project?

If you can provide a minimal reproduction with steps to reproduce, yes.

**Reply by bdim404** [reply, marked as answer]
> I also had this problem. With the way that you solved it did it require handling it within that getImageSlice.html? Would it be possible to add that update to the project?

Hi ! I found that the problem is `EXIF` message different with the picture look like, If you tyied to open the picture by [GIMP](https://www.gimp.org/), you will be notice that *"This image contains EXIF orientation metadata. Do you want to rotate the image?"*

So I write a python srcipt to fix that. You can try it from [here](https://gist.github.com/bdim404/9774a5e91e0311b185d7eafcb39a45a8).

**Reply by Tomp-Akoe** [reply]
> > I also had this problem. With the way that you solved it did it require handling it within that getImageSlice.html? Would it be possible to add that update to the project?
> 
> Hi ! I found that the problem is `EXIF` message different with the picture look like, If you tyied to open the picture by [GIMP](https://www.gimp.org/), you will be notice that _"This image contains EXIF orientation metadata. Do you want to rotate the image?"_
> 
> So I write a python srcipt to fix that. You can try it from [here](https://gist.github.com/bdim404/9774a5e91e0311b185d7eafcb39a45a8).

Hi, I tried it and it didn't solve the flip problem of the image. Can you take a look at your workflow?

**Reply by bdim404** [reply]
First of all, the script works perfectly fine for me without any problems. Secondly if you have any problems coming up, please let me know what the exact problem is. Lastly, it is recommended that the script be modified to suit your situation.
